File: frontend/src/api/strategy.ts
import { api } from './client';

export interface StrategyQuestionnaire {
  business_type: string;
  target_audience: string;
  goals: string[];
  platforms: string[];
  tone: string;
  posting_frequency: string;
  content_pillars?: string[];
}

export interface ContentPillar {
  name: string;
  description: string;
  percentage: number;
  sample_topics: string[];
}

export interface WeeklySlot {
  day_of_week: string;
  time: string;
  pillar: string;
  post_type: string;
}

export interface StrategyResponse {
  pillars: ContentPillar[];
  weekly_schedule: WeeklySlot[];
  post_ideas: Record<string, unknown>[];
  hashtag_strategy: Record<string, string[]>;
  growth_tactics: string[];
}

export const generateStrategy = (data: StrategyQuestionnaire) =>
  api.post<StrategyResponse>('/strategy/generate', data).then(r => r.data);

export const generatePostIdeas = (data: { strategy_context: string; count?: number; platform: string }) =>
  api.post<{ ideas: { caption: string; hashtags: string[]; post_type: string; platform: string }[] }>('/strategy/post-ideas', data).then(r => r.data);

export const analyzePerformance = (postData: Record<string, unknown>[]) =>
  api.post<{ insights: string[]; recommendations: string[]; top_performing_types: string[] }>('/strategy/analyze', { post_data: postData }).then(r => r.data);

File: backend/app/schemas/strategy.py
from pydantic import BaseModel, Field


class StrategyQuestionnaire(BaseModel):
    business_type: str = Field(
        ..., description="Type of business (e.g., 'SaaS startup', 'e-commerce', 'restaurant')"
    )
    target_audience: str = Field(
        ..., description="Description of the target audience"
    )
    goals: list[str] = Field(
        ..., description="Marketing goals (e.g., 'brand awareness', 'lead generation')"
    )
    platforms: list[str] = Field(
        ..., description="Social media platforms to target (e.g., 'instagram', 'twitter')"
    )
    tone: str = Field(
        ..., description="Brand voice/tone (e.g., 'professional', 'casual', 'witty')"
    )
    posting_frequency: str = Field(
        ..., description="Desired posting frequency (e.g., 'daily', '3x per week')"
    )
    content_pillars: list[str] | None = Field(
        default=None,
        description="Optional predefined content pillars; AI will suggest if not provided",
    )


class ContentPillar(BaseModel):
    name: str
    description: str
    percentage: float = Field(
        ..., ge=0, le=100, description="Percentage of total content allocated to this pillar"
    )
    sample_topics: list[str]


class WeeklySlot(BaseModel):
    day_of_week: str
    time: str
    pillar: str
    post_type: str


class StrategyResponse(BaseModel):
    pillars: list[ContentPillar]
    weekly_schedule: list[WeeklySlot]
    post_ideas: list[dict]
    hashtag_strategy: dict[str, list[str]]
    growth_tactics: list[str]


class PostIdeasRequest(BaseModel):
    strategy_context: str = Field(
        ..., description="Summary of the current content strategy for context"
    )
    count: int = Field(default=5, ge=1, le=20)
    platform: str


class PostIdea(BaseModel):
    caption: str
    hashtags: list[str]
    post_type: str
    platform: str


class PostIdeasResponse(BaseModel):
    ideas: list[PostIdea]


class PerformanceAnalysisRequest(BaseModel):
    post_data: list[dict] = Field(
        ..., description="List of posts with simplified metrics (likes, comments, shares, etc.)"
    )


class PerformanceAnalysisResponse(BaseModel):
    insights: list[str]
    recommendations: list[str]
    top_performing_types: list[str]
